pkg/image/api/types.go

```diff
@@ -60,6 +62,93 @@ type ImageLayer struct {
 	Size int64
 }
 
+const (
+	// The supported type of image signature.
+	ImageSignatureTypeAtomic string = "atomic"
+)
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// Conditions represent the latest available observations of a signature's current state.
+	Conditions []SignatureCondition
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed and
+	// the information available.
+
+	// A human readable string representing image's identity. It could contain company's name, a product name
+	// and a version. Eligible value is image's pull specification
+	// (e.g. "registry.access.redhat.com/rhel7/rhel:7.2").
+	ImageIdentity string
+	// Optional. Contains interesting signed claims not important enough to become primary fields.
```

Comment:
Rewritten.
On 06/06/2016 05:22 PM, David Eads wrote:
>
> In pkg/image/api/types.go <https://github.com/openshift/origin/pull/8371#discussion_r65910251>:
>
> > +type ImageSignature struct {
> > +	// Describes a type of stored blob.
> > +	Type string
> > +	// An opaque binary string which is an image's signature.
> > +	Content []byte
> > +	// Conditions represent the latest available observations of a signature's current state.
> > +	Conditions []SignatureCondition
> > +
> > +	// Following metadata fields will be set by server if the signature content is successfully parsed and
> > +	// the information available.
> > +
> > +	// A human readable string representing image's identity. It could contain company's name, a product name
> > +	// and a version. Eligible value is image's pull specification
> > +	// (e.g. "registry.access.redhat.com/rhel7/rhel:7.2").
> > +	ImageIdentity string
> > +	// Optional. Contains interesting signed claims not important enough to become primary fields.
>
> "Contains claims from the signature."
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/openshift/origin/pull/8371/files/4371c2172d2421e632faa67cf851acce66753c04#r65910251>, or mute the 
> thread <https://github.com/notifications/unsubscribe/ABaQk7vzEcArlzawtuh4lue1ArBl_ACtks5qJDtFgaJpZM4IBAFQ>.
>